src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/OptionsBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,312 @@
+                    .Validate(o => o.Boolean, "Boolean")
+                    .Validate(o => o.Integer > 12, "Integer")
+                    .ValidateOnStart();
+            });
+
+            using (var host = hostBuilder.Build())
+            {
+                var error = await Assert.ThrowsAsync<OptionsValidationException>(async () =>
+                {
+                    await host.StartAsync();
+                });
+
+                ValidateFailure<ComplexOptions>(error, 2, "Boolean", "Integer");
+            }
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        private async Task ValidateOnStart_AddBothLazyAndEagerValidationOnSameType_SkipsLazyValidationWhenHostStarts()
+        {
+            var hostBuilder = CreateHostBuilder(services =>
+            {
+                // Adds eager validation using ValidateOnStart
+                services.AddOptions<ComplexOptions>("correct_configuration")
+                    .Configure(o => o.Boolean = true)
+                    .Validate(o => o.Boolean, "correct_configuration")
+                    .ValidateOnStart();
+
+                // Adds lazy validation, skipping validation on start
+                services.AddOptions<ComplexOptions>("bad_configuration")
+                    .Configure(o => o.Boolean = false)
+                    .Validate(o => o.Boolean, "bad_configuration");
+            });
+
+            // For the lazily added "bad_configuration", validation failure does not occur when host starts
+            using (var host = hostBuilder.Build())
+            {
+                try
+                {
+                    await host.StartAsync();
+                }
+    #pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception ex)
+    #pragma warning restore CA1031 // Do not catch general exception types
+                {
+                    Assert.True(false, "Expected no exception, but got: " + ex.Message);
+                }
+            }
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        public async Task ValidateOnStart_AddBothLazyAndEagerValidationOnDifferentTypes_SkipsLazyValidationWhenHostStarts()
+        {
+            var hostBuilder = CreateHostBuilder(services =>
+            {
+                // Lazy validation
+                services.AddOptions<NestedOptions>()
+                    .Configure(o => o.Integer = 11)
+                    .Validate(o => o.Integer > 12, "Integer");
+
+                // Eager validation
+                services.AddOptions<ComplexOptions>()
+                    .Configure(o => o.Boolean = false)
+                    .Validate(o => o.Boolean, "first Boolean must be true.")
+                    .ValidateOnStart();
```

Comment:
Do we have any tests that call `.ValidateOnStart()` with valid options and make sure no errors are raised?

Reply:
Since the tests are now sectioned into two different places, I'm gonna look back at the `..Hosting/tests/UnitTests/OptionsBuilderExtensionsTests.cs` test file and make sure it contains all the unit tests we want and leave the other one in `..Options.Tests/OptionsBuilderTest.cs` to focus solely on DataAnnotations.